[issues/678] update pom.xml accordingly

bigmarvin opened the pull request:
To address https://github.com/prometheus/client_java/issues/678 --

1. make the artifacts of name `simpleclient_tracer_*` packed as bundles
2. make the resolution of imported packages optional by default

@fstab and @tomwilkie, may I ask you maintainers to take a look?

If possible, I'm also requesting the delivery of this fix in 0.11.x asap, as the absence of must-import packages prevents the upgrade of `simpleclient` in OSGi runtime.

fstab commented:
Oh, I fixed part of #678, and it seems I have overlooked that there is a PR for that. The current status on `master` is: I added `<packaging>bundle</packaging>`, but I'm unsure what `*;resolution:=optional` does, so I didn't add that. I guess some dependencies are not optional. Might it break things if we declare everything optional? I'm not an OSGI user, so I could use some help explaining the effect of `*;resolution:=optional`.

bigmarvin commented:
> Oh, I fixed part of #678, and it seems I have overlooked that there is a PR for that. The current status on `master` is: I added `<packaging>bundle</packaging>`, but I'm unsure what `*;resolution:=optional` does, so I didn't add that. I guess some dependencies are not optional. Might it break things if we declare everything optional? I'm not an OSGI user, so I could use some help explaining the effect of `*;resolution:=optional`.

Great, thanks for the efforts on this! Let's continue the discussion of 2nd part in #678.
